openreview/tools.py

```diff
@@ -15,6 +15,13 @@
 import urllib.parse as urlparse
 from tqdm import tqdm
 from concurrent.futures import ThreadPoolExecutor
+import json
+import os
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+with open(os.path.join(dir_path, 'duplicate_domains.json')) as f:
+    duplicate_domains = json.load(f)
```

Comment:
not every load, only at the `import tools` which happen only once at the start. I don't think it would be a problem, but I can change it load and pass it to the subdomains function call.

Reply:
Yes, that's what I mean. So, for example, if there are two files with `import tools`, will it load the json file twice or just once? So what we want is to load the file only once, and preferably only when the method to compute conflicts is called.

Reply:
I see, yes, that works, but can we make it so that it's only loaded once when we compute the conflicts? I say this, because it's a bit strange to instantiate a client and load the file when it's not needed, don't you think?

Reply:
No, not really. Loading the file when only needed will actually make it less efficient as it will load the file for every call, while it only loads it once now. I can move the loading of the JSON to another file and import the dictionary from there but that is just an indirect way of doing what's happening now. I am happy to go with either version or if you have any other suggestion.

Reply:
How about this: You can load it in the function by checking if it has not been loaded yet. If it has, you don't need to do it again. You could even have a loader function that does this. If we are eventually moving this to the database, we would not want to query the database as soon as we instantiate the client. You could just modify the loader function then, and the request will only be made once. What do you think?

Reply:
Another option is go to the caller of subdomains when conflicts are computed and load it there and then pass it as a parameter to the subdomains function.

Reply:
both options require maintaining a global variable which again more or less is just an indirect way of doing the same thing.

I don't think I am getting why it's bad to load it at the start, but I have separated the loading JSON part into another file and added the import inside the `subdomains` function.

Reply:
Thanks for the changes. I think this is what we want in a way. I'm not an expert in python, but have you considered using decorators? I found an example that may be useful for what we want: https://stackoverflow.com/a/4104188.
Let me know what you think

Reply:
Yes, it could work as well. It just seems a fancier way to do the same thing.

Reply:
I think it's cleaner (we don't need the `common.py` file and avoid using import inside the function) and the implementation can be changed in the future very easily to load from a different datasource.